e2e/ansible/playbooks/hyperconverged/test-k8s-pvc-deletion/k8s_delete_pvc.yml

```diff
@@ -0,0 +1,147 @@
+- hosts: localhost
+
+  vars_files:
+    - k8s-delete-pvc-vars.yml
+
+  tasks:
+     
+       - name: Get the number of nodes in the cluster
+         shell: kubectl get nodes | grep '<none>' | wc -l
+         args:
+           executable: /bin/bash
+         register: node_out
+         delegate_to: "{{groups['kubernetes-kubemasters'].0}}"
+
+       - name: Fetch the node count from stdout
+         set_fact:
+            node_count: " {{ node_out.stdout}}"
+
+       - name: Replace the replica count in storage class file
+         replace:
+           path: storage-class.yaml 
+           regexp: 'openebs.io/jiva-replica-count: "1"'
+           replace: 'openebs.io/jiva-replica-count: "{{ (node_count) |int+1}}"'
+           backup: yes
+
+       - name: Get $HOME of K8s master for kubernetes user
+         shell: source ~/.profile; echo $HOME
+         args:
+           executable: /bin/bash
+         register: result_kube_home
+         delegate_to: "{{groups['kubernetes-kubemasters'].0}}"
+
+       - name: Copy the volume claim to k8s master
+         copy:
+           src: "{{ sc_def }}"
+           dest: "{{ result_kube_home.stdout }}"
+         delegate_to: "{{ groups['kubernetes-kubemasters'].0 }}"
+
+       - name: Check whether maya-apiserver pod is deployed
+         shell: source ~/.profile; kubectl get pods | grep maya-apiserver
+         args:
+           executable: /bin/bash
+         register: result
+         delegate_to: "{{groups['kubernetes-kubemasters'].0}}"
+         until: "'Running' in result.stdout"
+         delay: 120
+         retries: 5
+
+       - name: Create the storage class in K8s master
+         shell: kubectl apply -f "{{ sc_def }}"
+         args:
+           executable: /bin/bash
+         delegate_to: "{{ groups['kubernetes-kubemasters'].0 }}"
+
+       - name: Copy pvc yaml to K8s master
+         copy:
+           src: "{{ pvc_def }}"
+           dest: "{{ result_kube_home.stdout }}"
+         delegate_to: "{{ groups['kubernetes-kubemasters'].0 }}"
+      
+       - name: Create PVC in k8s cluster
+         shell: kubectl apply -f "{{ pvc_def }}"
+         args:
+           executable: /bin/bash
+         delegate_to: "{{ groups['kubernetes-kubemasters'].0 }}"
+      
+       - name: Check if the svc is created
+         shell: kubectl get svc | grep pvc | grep ctrl
+         args:
+           executable: /bin/bash
+         register: svc
+         delegate_to: "{{ groups['kubernetes-kubemasters'].0 }}" 
+         until: "'<none>' in svc.stdout"
+         delay: 100
+         retries: 3      
```

Comment:
- Let's check against the PV name which can be derived after deploying create_pvc.yaml (there may be residues from previous tests)
- The service gets created almost instantaneously after deploying. Better would be carry out this check after pod creation is successful, i.e., replica and controller rollout is complete (there is ideally no need to have a 100s delay b/w retries)